urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/ui/AnimatedPermFrame.java

```diff
@@ -0,0 +1,669 @@
+
+/*
+ * AnimatedPermFrame.java
+ *
+ * Created on Apr 2, 2012, 2:08:31 PM
+ */
+package permlab.ui;
+
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.KeyboardFocusManager;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageTypeSpecifier;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.metadata.IIOMetadata;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JMenuItem;
+import javax.swing.JPanel;
+import javax.swing.Timer;
+import permlib.Permutation;
+import permlib.property.PermProperty;
+import permlib.property.Universal;
+import permlab.utilities.ExportInterface;
+import static permlab.utilities.ExportInterface.fileChooser;
+import permlab.utilities.FileUtilities;
+import permlab.utilities.PaintUtilities;
+
+/**
+ * The GUI class that represents the frame for displaying an animation of a
+ * series of permutations.
+ *
+ * @author Michael Albert, M Belton
+ */
+public class AnimatedPermFrame extends javax.swing.JFrame {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 61 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.